6-AdvancedScenarios/3-call-api-acrs/README.md

```diff
@@ -0,0 +1,345 @@
+
+1. Open the `SPA/src/authConfig.js` file.
+1. Find the string `Enter_the_Application_Id_Here` and replace it with the application ID (clientId) of `msal-react-spa` app copied from the Azure portal.
+1. Find the string `Enter_the_Tenant_Info_Here` and replace it with your Azure AD tenant ID.
+1. Find the string `Enter_the_Web_Api_Scope_here` and replace it with the scope of `msal-node-api` that you've exposed earlier, e.g. `api://API_CLIENT_ID/access_as_user`.
+
+## Running the sample
+
+Start the web API first:
+
+```console
+    cd API
+    npm start
+```
+
+In a separate terminal, type
+
+```console
+    cd SPA
+    npm start
+```
+
+## Explore the sample
+
+1. We'd first replicate the experience of an admin configuring the auth contexts. For that, browse to `http://localhost:5000/admin` and sign-in using a tenant admin account. Click on the **Dashboard** link on the menu.
+
+![Overview](./ReadmeFiles/Admin.png)
+
+2. As a first step, you will ensure that a set of auth contexts is already available in this tenant. Click the **Create or Fetch** button to check if they exist. If they don't, the code will create three example auth context entries for you. These three entires are named `Require strong authentication`, `Require compliant devices` and `Require trusted locations`.
+
+> :information_source: The Graph permission **Policy.ReadWrite.ConditionalAccess** is required for creating new records. In production, the permission **Policy.Read.ConditionalAccess** should be sufficient to read existing values and thus is recommended.
+
+![Overview](./ReadmeFiles/Create-Fetch_Click.png)
+
+Select an operation in the web API and an auth context value to apply. Then select **Save or Update**. This updates this mapping in the local app's database. We advise you use the same auth context value for operations if possible, as this ensures that the user is redirected to Azure AD just once to perform the step-up authN.
+
+> :warning: When changing auth context mappings, have the user sign-out and sign-back in for the changes to take effect.
+
+1. Go to **Details** page to get details of data saved on the web API side in its database. You can select **Delete** if you need to delete a mapping from the local database.
+
+![Overview](./ReadmeFiles/ViewDetails.png)
+
+The web API is now ready to challenge users for step-up auth for the selected operations.
+
+### Configure a conditional access policy to use auth context in Azure portal
+
+1. Navigate to **Azure Active Directory** > **Security** > **Conditional Access**
+1. Select **New policy** and go to **Cloud apps or actions**. In dropdown select **Authentication context**. The newly created auth context values will be listed for you to be used in this CA policy.
+
+![Overview](./ReadmeFiles/AuthContext.png)
+
+Select the value and create the policy as required. For example, you might want the user to satisfy a **MFA** challenge if the auth context value is **c1**.
```

Comment:
>c1 [](http://example.com/codeflow?start=151&length=2)

Avoid using c1 and c2, instead, us the auth context name instead

Reply:
changed!